com.unity.netcode.gameobjects/Runtime/com.unity.netcode.runtime.asmdef

```diff
@@ -2,11 +2,11 @@
     "name": "Unity.Netcode.Runtime",
     "rootNamespace": "Unity.Netcode",
     "references": [
-        "Unity.Multiplayer.MetricTypes",
-        "Unity.Multiplayer.NetStats",
-        "Unity.Multiplayer.NetStatsReporting",
-        "Unity.Multiplayer.NetworkSolutionInterface",
-        "Unity.Collections"
+        "GUID:7c4416f9e8f132246b0e1c1ec45be7d6",
+        "GUID:b1cd7326e664a434ab35daa802773c7f",
+        "GUID:821943bbff0df3b48b43584136c29e17",
+        "GUID:272061007ed268345bb3d3df8b09b583",
+        "GUID:e0cd26848372d4e5c891c569017e11f1"
```

Comment:
my feedback would be to keep non-GUID references around and add `x.Tools.y` in the middle of asmdef references.

Reply:
Why's that Fatih? The nice thing about the GUIDs is that this change can be made to NGO with GUIDs will be compatible with both the old and the new tools assembly names, whereas if we just switch the name then NGO will become incompatible with older versions of tools that still have the old name.

Reply:
|                                        | **NGO With Old Names** | **NGO With New Names** | **NGO With Guids**|
| ---------------              | ---------------             | ---------------              |--------------- |
| **Tools With Old Names**  | Compatible                   | Incompatible                 | Compatible       |
| **Tools with New Names** | Incompatible                 | Compatible                    | Compatible      |

Reply:
At least on the tools team all of our external dependency assembly references are GUIDs so that if any of the assemblies of our external dependencies is renamed it won't break our build or result in additional incompatibilities between versions.

Reply:
It depends on which contract we'd like to rely on: by name, by GUID.
By GUID, we're relying on a very specific version of a file, the asset GUID of an asmdef asset.
By name, we're relying on the name/namespace of the asmdef and that gives us an opportunity to swap assemblies of an asmdef without breaking anything as long as the name is the same (this would sound like non-issue, a minor pro until it's not, being able to do this would save the day some day — and it's not really good to manually copy-paste asset GUIDs around to keep "compatibility" around, you might bump into GUID conflicts some day. another possibility would be having different asmdefs with the same names on different target platforms to have multi-assembly setups etc).
There's also readability aspect of names compared against GUIDs: better tracking, debugging, maintaining of asmdef dependencies.

FWIW, you can take a look at other packages we have in Unity and observe that we generally favor name refs instead of GUID refs:
- [com.unity.physics/Unity.Physics/Unity.Physics.asmdef](https://github.cds.internal.unity3d.com/unity/dots/blob/master/Packages/com.unity.physics/Unity.Physics/Unity.Physics.asmdef)
- [com.unity.jobs/Unity.Jobs/Unity.Jobs.asmdef](https://github.cds.internal.unity3d.com/unity/dots/blob/master/Packages/com.unity.jobs/Unity.Jobs/Unity.Jobs.asmdef)
- [com.unity.collections/Unity.Collections/Unity.Collections.asmdef](https://github.cds.internal.unity3d.com/unity/dots/blob/master/Packages/com.unity.collections/Unity.Collections/Unity.Collections.asmdef)
- [com.unity.entities/Unity.Entities/Unity.Entities.asmdef](https://github.cds.internal.unity3d.com/unity/dots/blob/master/Packages/com.unity.entities/Unity.Entities/Unity.Entities.asmdef)
- [com.unity.animation/Unity.Animation/Unity.Animation.asmdef](https://github.cds.internal.unity3d.com/unity/dots/blob/master/Packages/com.unity.animation/Unity.Animation/Unity.Animation.asmdef)
- etc.

Having said that, it's ultimately Tools team's decision to user GUID-based or name-based asmdef references internally in tools packages. On our side, we'd rather stick with name-based asmdef references as much as we can.

> if we just switch to the new names in NGO then NGO will become incompatible with older versions of tools

I see the point but it begs the question how far an old version of Tools package would be able to keep up with newer versions of the NGO package? I mean, how realistic it is to say "NGO + 10 versions will still be compatible with the same Tools package version"?

If you guys would like to proceed with GUID-based refs everywhere, maybe the compromise from NGO side for Tools package here would be something like this:
```diff
    "references": [
// use GUID refs for Tools
-        "Unity.Multiplayer.MetricTypes",
-        "Unity.Multiplayer.NetStats",
-        "Unity.Multiplayer.NetStatsReporting",
-        "Unity.Multiplayer.NetworkSolutionInterface",
+        "GUID:7c4416f9e8f132246b0e1c1ec45be7d6",
+        "GUID:b1cd7326e664a434ab35daa802773c7f",
+        "GUID:821943bbff0df3b48b43584136c29e17",
+        "GUID:272061007ed268345bb3d3df8b09b583",
        "Unity.Collections" // keep other refs as they are
],
```

Oh, maybe a last (more subjective, personal) note that I see `Unity.Netcode`, `Unity.Jobs` etc names changing less frequently (if ever once decided) than actual asset files. If we keep changing our names, we might have some other issues somewhere else.

Reply:
If you guys prefer names, we can definitely use names. There are many advantages especially the readability.

Part of my hesitation to introduce any incompatibilities between versions is that I don't know what versions we're intending to maintain compatibility between. For that reason, I'm going to wait until I have the sign off of Beck or someone else who knows the answer to this before I submit these changes.